examples/pipeline_wav2letter/main.py

```diff
@@ -0,0 +1,668 @@
+import argparse
+import logging
+import os
+import string
+from datetime import datetime
+from time import time
+
+import torch
+import torchaudio
+from torch.optim import SGD, Adadelta, Adam, AdamW
+from torch.optim.lr_scheduler import ExponentialLR, ReduceLROnPlateau
+from torch.utils.data import DataLoader
+from torchaudio.datasets.utils import bg_iterator
+from torchaudio.models.wav2letter import Wav2Letter
+
+from ctc_decoders import GreedyDecoder
+from datasets import collate_factory, split_process_librispeech
+from languagemodels import LanguageModel
+from metrics import levenshtein_distance
+from transforms import Normalize, UnsqueezeFirst
+from utils import MetricLogger, count_parameters, save_checkpoint
+
+
+def parse_args():
```

Comment:
I'd specialize this file to doing just this and split out everything else into another file. You could call it "engine" or such.

Reply:
That way you're separating this way of interfacing with your program (i.e. the CLI) from the actual logic.